feat: burner wallets deployer contract

remybar opened the pull request:
## Description

The `BurnerWalletDeployer` contract allows to deploy OZ accounts to be used as burner wallets.

This is done with the `deploy_burner_wallet` which takes the account public key as parameter and returns the created account address.

To not deploy a same account forever, this contract uses an internal `salt` starting from 0 and increased of 1 after each new burner wallet.

## Additional Information

- [x] I have read the [contributing docs](../CONTRIBUTING.md) (if this is your first contribution)
- [x] This is not a duplicate of any [existing pull request](https://github.com/argentlabs/Starknet-Scaffold/pulls)

## Related Issues

_Closes #27_

remybar commented:
Hi @Darlington02 

I've tested this burner wallet deployer contract in local by declaring the OZ preset account and used this local classhash for my tests. 

Do you want me to add some unit tests for this contract ?

Cheelax commented:
GM ser! Thank you for the PR!
I will take the frontend part of this issue.
Do you have any advice on how to call it from the typescript side?

Darlington02 commented:
@remybar yes please, ensure to add good test coverage. And please assist @Cheelax with advice on how to go about the frontend integration:)

remybar commented:
> GM ser! Thank you for the PR! I will take the frontend part of this issue. Do you have any advice on how to call it from the typescript side?

Hi @Cheelax !
Sure, tomorrow I can explain you how to call a contract from the front-end side. I will add a comment to your issue so if you have some questions we will be able to discuss at the right place ;-)

remybar commented:
Hi @Darlington02 

I've added some tests with snforge but:
- I was not able to compile/run these tests with Cairo 2.5.3 and snfoundry 0.18.0. I got a weird error: `missing field `test_details` at line 1 column 269339`. I asked the question on Starknet discord and I wait for an answer. I will update my PR as it would be better to use the last Cairo version.
- I'm not familiar with Cairo smart contract testing yet. I wanted to do some things but I failed:
  - At first, I wanted to mock `deploy_syscall` to just check that this function was correctly called (so, a basic unit test). But I didn't find if it is possible to do such kind of mocks and how 😅 
  - Next, I wanted to directly use the preset `Account` done by OpenZeppelin in my tests but I was not able to use it with snforge. I added OZ as a dev-dependencies in Scarb.toml but snforge said that `openzeppelin` was not there ...
  - Next, I decided to just create a simple mock `Account` but I wanted to declare it with the same class hash as the OZ account hash thanks to snforge, so I would have been able to keep it as a constant in the smart contract. I found `deployedAt` to deploy at an address but not something like `declareAt` 😅 
  - At the end, I kept my mock `Account` and modified the `BurnerWalletDeployer` constructor to be able to provide a specific classhash. If 0, it uses the OZ class hash by default.

Anyway, if you have some advices to improve these tests, I will strongly appreciate them :)

Darlington02 commented:
Hi @remybar firstly, amazing work you did. Had to merge, as I have no actions setup to run tests on PR.

I’d take a look and fix the tests from my end:)

but the contract looks perfect 👍

remybar commented:
> Hi @remybar firstly, amazing work you did. Had to merge, as I have no actions setup to run tests on PR.
> 
> I’d take a look and fix the tests from my end:)
> 
> but the contract looks perfect 👍

Thanks ! 🙏 

Could you ping me if you change the tests ? I would be very interesting to see how to improve them ;-) Thanks 👍

Darlington02 commented:
Sure, I will:)
